packages/opencensus-propagation-binaryformat/src/util.ts

```diff
@@ -0,0 +1,100 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This util contains the functions for serializing and deserializing
+ * SpanContext using a binary format.
+ *
+ * See: https://github.com/census-instrumentation/opencensus-specs/blob/
+ *       master/encodings/BinaryEncoding.md
+ */
+
+import {SpanContext} from '@opencensus/core';
+import {decToHex, hexToDec} from 'hex2dec';
+
+const VERSION_ID = 0;
+const TRACE_ID_FIELD_ID = 0;
+const SPAN_ID_FIELD_ID = 1;
+const TRACE_OPTION_FIELD_ID = 2;
+
+// Sizes are number of bytes.
+const ID_SIZE = 1;
+const TRACE_ID_SIZE = 16;
+const SPAN_ID_SIZE = 8;
+const TRACE_OPTION_SIZE = 1;
+
+const FORMAT_LENGTH =
+    4 * ID_SIZE + TRACE_ID_SIZE + SPAN_ID_SIZE + TRACE_OPTION_SIZE;
+
+/**
+ * Serialize the given span context into a Buffer.
+ * @param spanContext The span context to serialize.
+ */
+export function serializeSpanContext(spanContext: SpanContext): Buffer {
+  /**
+   *  0           1           2
+   *  0 1 2345678901234567 8 90123456 7 8
+   * -------------------------------------
+   * | | |                | |        | | |
+   * -------------------------------------
+   *  ^ ^      ^           ^    ^     ^ ^
+   *  | |      |           |    |     | `-- options value (spanContext.options)
+   *  | |      |           |    |     `---- options field ID (2)
+   *  | |      |           |    `---------- spanID value (spanContext.spanID)
+   *  | |      |           `--------------- spanID field ID (1)
+   *  | |      `--------------------------- traceID value (spanContext.traceID)
+   *  | `---------------------------------- traceID field ID (0)
+   *  `------------------------------------ version (0)
+   */
+  const result = Buffer.alloc(FORMAT_LENGTH, 0);
+  result.write(spanContext.traceId, 2, TRACE_ID_SIZE, 'hex');
+  result.writeUInt8(SPAN_ID_FIELD_ID, 18);
+  // Convert Span ID from decimal to base 16 representation, then left pad into
+  // a length-16 hex string. (decToHex prepends its output with '0x', so we
+  // also slice that off.)
+  const base16SpanId =
+      `0000000000000000${decToHex(spanContext.spanId).slice(2)}`.slice(-16);
+  result.write(base16SpanId, 19, 8, 'hex');
+  result.writeUInt8(TRACE_OPTION_FIELD_ID, 27);
+  result.writeUInt8(spanContext.options || 0, 28);
+  return result;
+}
+
+/**
+ * Deseralize the given span context from binary encoding. If the input is a
+ * Buffer of incorrect size or unexpected fields, then this function will return
+ * null.
+ * @param buffer The span context to deserialize.
+ */
+export function deserializeSpanContext(buffer: Buffer): SpanContext|null {
+  const result: SpanContext = {traceId: '', spanId: ''};
+  // Length must be 29.
+  if (buffer.length !== FORMAT_LENGTH) {
+    return null;
+  }
+  // Check version and field numbers.
+  if (buffer.readUInt8(0) !== VERSION_ID ||
```

Comment:
Consider using `offset` constants instead of hard coding them here. Example in Java: https://github.com/census-instrumentation/opencensus-java/blob/0f8b8215ed2225b5118b0bda1d10af30037e3e3b/impl_core/src/main/java/io/opencensus/implcore/trace/propagation/BinaryFormatImpl.java#L78.

Reply:
Done.